File: CharactorSelectorApi/SurveyApi/Models/Entities/Survey.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace SurveyApi.Models.Entities
{
    /// <summary>
    /// This is the entity for Survey.
    /// Survey to Questions are 1-to-many.
    /// </summary>
    public class Survey
    {
        [Key]
        public Guid Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public int Type { get; set; }

        /// <summary>
        /// The author of this survey.
        /// </summary>
        public Guid? UserId { get; set; }

        public string UserName { get; set; }
        public DateTime CreatedDateUtc { get; set; }
        public DateTime UpdatedDateUtc { get; set; }

        public virtual List<Question> Questions { get; set; }
    }
}

File: survey-web/src/app/Models/survey.ts
import {Option} from './option';
import {Time} from '@angular/common';
import {Question} from './question';

export interface Survey {
    id: string;
    name: string;
    description: string;
    type: string;
    userId: string;
    userName: string;
    createdDateUtc: Time;
    updatedDateUtc: Time;
    questions: Question[];
}
